atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventCacheIntegrationTest.java

```diff
@@ -391,6 +398,28 @@ public void newEventsCauseOldEventsToBeDeleted() {
         verifyStage();
     }
 
+    @Test
+    public void commitLocksAreCorrectlyFilteredOutUsingServerToken() {
+        setupInitialState();
+        eventCache.processStartTransactionsUpdate(ImmutableSet.of(), SUCCESS);
+
+        // simulates the actual lock token that the client receives
+        ConjureLockToken serverToken = ConjureLockToken.of(COMMIT_TOKEN.getRequestId());
+        LockToken commitToken = LeasedLockTokenCreator.of(
+                serverToken,
+                Lease.of(LeaderTime.of(LeadershipId.random(), NanoTime.createForTests(1L)), Duration.ZERO));
+        eventCache.processGetCommitTimestampsUpdate(
+                ImmutableSet.of(ImmutableTransactionUpdate.builder()
+                        .commitTs(5L)
+                        .startTs(START_TS)
+                        .writesToken(commitToken)
+                        .build()),
+                SUCCESS_2);
+
+        assertThat(eventCache.getCommitUpdate(START_TS).accept(new CommitUpdateVisitor()))
+                .containsExactlyInAnyOrder(DESCRIPTOR);
```

Comment:
this shows that `LOCK_EVENT` is filtered out, which contains `DESCRIPTOR_3`, whereas `LOCK_EVENT_2` is present, with `DESCRIPTOR` present.